### PR TITLE
destructure LessonNextSteps for versatility

### PR DESCRIPTION
**Pull request by joelhooks**
This breaks LessonNextSteps apart so that the status of a lesson is its own component, and next steps is strictly concerned with actions. Also added some colors for buttons and modified some text.

The LessonTeaser needs lots of design love. Was thinking about that here: https://paper.dropbox.com/doc/Lists-of-Lessons-LOL-hkj93JkuRH5ggz45BHrvJ

image

image

### CODE REVIEW

File: src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import {Link} from 'react-router-dom'
 import {Heading} from 'egghead-ui'
```
Comment (joelhooks) [inline src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js:3]: `LessonTeaser` is probably a poor name at this point 😄

File: src/components/LessonNextSteps/index.js [hunk 2]
```diff
@@ -31,6 +23,7 @@ export default ({lesson}) => (
                     <Button
                       size='extra-small'
                       onClick={() => request()}
+                      color={colorByLessonState[state].color}
```
Comment (joelhooks) [inline src/components/LessonNextSteps/index.js:26]: mostly to mark destructive actions in red, they still need a confirmation.

File: src/components/LessonStatus/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import {statusByLessonState} from 'utils/lessonStates'
+
+export default function LessonStatus({lesson}) {
```
Comment (joelhooks) [inline src/components/LessonStatus/index.js:4]: Wanted this separated from the actions, mostly for layout flexibility.

File: src/components/LessonUploadEdit/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import {
+  uploadVideoTitleText,
+  editLessonTitleText,
+  replaceVideoTitleText,
+} from 'utils/text'
+import Anchor from 'components/Anchor'
+import {Button} from 'egghead-ui'
+
+export default function LessonUploadEdit ({lesson}) {
```
Comment (joelhooks) [inline src/components/LessonUploadEdit/index.js:10]: upload and edit button to slap around as needed

File: src/components/LessonUploadEdit/index.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <span>
+    <div className='flex mb3'>
+    {
+      lesson.edit_lesson_http_url ?
+        <div className="pa1">
```
Comment (joelhooks) [inline src/components/LessonUploadEdit/index.js:16]: playing ball with wrappers 👍

File: src/utils/lessonStates/index.js [hunk 6]
```diff
@@ -142,3 +142,42 @@ export const actionByLessonState = {
     title: retireActionText,
   },
 }
+
+export const colorByLessonState = {
```
Comment (joelhooks) [inline src/utils/lessonStates/index.js:146]: 🌈

File: src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js [hunk 7]
```diff
@@ -31,9 +33,13 @@ export default ({instructor, lesson}) => {
             : null
           }
         </div>
+        <LessonStatus lesson={lesson}/>
+        <div className="mt3">
+        <LessonUploadEdit lesson={lesson}/>
```
Comment (tayiorbeii) [inline src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js:38]: indent?

File: src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js [hunk 8]
```diff
@@ -31,9 +33,13 @@ export default ({instructor, lesson}) => {
...
+        </div>
+        <div className="mt1">
         <LessonNextSteps lesson={lesson}/>
```
Comment (tayiorbeii) [inline src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js:41]: indent?

**Review by tayiorbeii — APPROVED** [review]

File: src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js (hunk 1 above)
Comment (trevordmiller) [reply src/components/LessonList/components/PaginatedLessonList/components/LessonTeaser/index.js:3]: Could be `LessonListItem`

File: src/components/LessonUploadEdit/index.js (hunk 5 above)
Comment (trevordmiller) [reply src/components/LessonUploadEdit/index.js:16]: Thanks :)

**Review by trevordmiller — APPROVED** [review]
Cool beans

**Pull request merged**